app/adapters/alma_adapter/availability_status.rb

```diff
@@ -64,6 +64,12 @@ def holding_status(holding:, sequence:)
         # Notice that we only check if a holding is available via CDL when necessary
         # because it requires an extra (slow-ish) API call.
         status[:cdl] = cdl_holding?(holding["holding_id"])
+      elsif status[:status_label] == "Some items not available"
+        # Alma gives an incorrect status when records have no items. Here we check
+        # if that is the case and return a more accurate status. This logic would be
+        # better inside the Status class but we keep it here since it requires an
+        # extra API call to get the item data.
+        status[:status_label] = "On-site access" if item_data.empty?
```

Comment:
If `item_data` is NOT empty, then the status label is is nil. Is this what we want?

Reply:
Won't it still be "Some items not available"?

Reply:
As @hackartisan said, it will only change the value if there are no items, and that is the intended behavior. 

These two tests shows it in action: https://github.com/pulibrary/bibdata/blob/issue_1314_onsite/spec/adapters/alma_adapter_spec.rb#L247-L257